src/Note.cpp

```diff
@@ -0,0 +1,378 @@
+            }
+            count++;
+        }
+    }
+    return nullptr;
+}
+
+void Notes::save(FILE *file) const {
+    for (auto &note : notes_)
+        note.save(file);
+}
+
+static Note &ensure_note(Char &ch) {
+    if (!ch.pnote)
+        ch.pnote = std::make_unique<Note>(ch.name);
+    return *ch.pnote;
+}
+
+Note Note::from_file(FILE *fp) {
+    auto expect = [&](std::string_view expected) {
+        auto *word = fread_word(fp);
+        if (!matches(word, expected))
+            throw std::runtime_error(fmt::format("Expected '{}'", expected));
+    };
+    auto expect_str = [&](std::string_view expected) {
+        expect(expected);
+        return fread_stdstring(fp);
+    };
+    Note note(expect_str("sender"));
+    note.date_ = expect_str("date");
+    expect("stamp");
+    note.date_stamp_ = Clock::from_time_t(fread_number(fp));
+    note.to_list_ = expect_str("to");
+    note.subject_ = expect_str("subject");
+    note.text_ = expect_str("text");
+    return note;
+}
+
+void Note::subject(std::string_view subject) { subject_ = subject; }
+void Note::date(Time point) {
+    date_ = formatted_time(point);
+    date_stamp_ = point;
+}
+void Note::to_list(std::string_view to_list) { to_list_ = to_list; }
+void Note::add_line(std::string_view line) { text_ += fmt::format("{}\n\r", line); }
+void Note::remove_line() { text_ = remove_last_line(text_); }
+
+NoteHandler &NoteHandler::singleton() {
+    static auto on_change = [](NoteHandler &handler) {
+        if (auto *file = fopen(NOTE_FILE, "w")) {
+            handler.write_to(file);
+            fclose(file);
+        } else {
+            perror(NOTE_FILE);
+        }
+    };
+    static NoteHandler singleton(on_change);
+    return singleton;
+}
+
+void note_initialise() {
+    auto &handler = NoteHandler::singleton();
+    if (auto *fp = fopen(NOTE_FILE, "r")) {
+        handler.read_from(fp);
+        fclose(fp);
+    }
+}
+
+void do_note(Char *ch, const char *argument) {
+    if (ch->is_npc()) {
+        return;
+    }
+    char arg[MAX_INPUT_LENGTH];
+    auto note_remainder = smash_tilde(one_argument(argument, arg));
+
+    auto &handler = NoteHandler::singleton();
+    handler.on_command(*ch, ArgParser(argument));
+}
+
+NoteHandler::NoteHandler(OnChangeFunc on_change_func) : on_change_func_(std::move(on_change_func)) {
+    sub_commands.add("read", &NoteHandler::read);
+    sub_commands.add("list", &NoteHandler::list);
+    sub_commands.add("+", &NoteHandler::add_line);
+    sub_commands.add("-", &NoteHandler::remove_line);
+    sub_commands.add("subject", &NoteHandler::subject);
+    sub_commands.add("to", &NoteHandler::to);
+    sub_commands.add("clear", &NoteHandler::clear);
+    sub_commands.add("show", &NoteHandler::show);
+    sub_commands.add("post", &NoteHandler::post);
+    sub_commands.add("send", &NoteHandler::post);
+    sub_commands.add("remove", &NoteHandler::remove);
+    sub_commands.add("delete", &NoteHandler::delet, MAX_LEVEL - 2);
```

Comment:
hah `delet`

Reply:
yeah sucks, right? The worst one to hit is `register` as we've not really used that as a keyword in anger since the early 90s